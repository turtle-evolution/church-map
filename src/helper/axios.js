import axios from 'axios';

import history from '../services/helper/history';
import jwtHelpers from '../services/helper/jwtHelpers';

import environmentConstants from './constants/environmentConstants';
import NotFoundUrlList from './constants/NotFoundUrlList';
import UnauthorizedUrlList from './constants/UnauthorizedUrlList';

const httpHandler = axios.create({
  baseURL: environmentConstants.ROOT_URL
});

const RETRY_SYMBOL = Symbol('retry');

httpHandler.interceptors.request.use(
  function intercept(config) {
    return {
      ...config,
      withCredentials: 'true'
    };
  },
  function interceptError(error) {
    return Promise.reject(error);
  }
);

httpHandler.interceptors.response.use(
  function intercept(response) {
    return response;
  },
  function interceptError(error) {
    const originalRequest = error.config;

    switch (error.response.status) {
      case 401:
        if (originalRequest.url.includes(UnauthorizedUrlList)) {
          return Promise.reject(error);
        }
        return history.push('/sign-out');

      case 404:
        if (originalRequest.url.includes(NotFoundUrlList)) {
          return Promise.reject(error);
        }
        return history.push('/not-found');

      case 403:
        if (originalRequest.url === `/api/users/refresh-token`) {
          jwtHelpers.removeRefreshToken();
          history.push('/');
          return Promise.reject(error);
        }
        if (!originalRequest[RETRY_SYMBOL]) {
          originalRequest[RETRY_SYMBOL] = true;
          const refreshToken = jwtHelpers.getRefreshToken();
          return httpHandler
            .post('/users/refresh-token', refreshToken, {
              headers: {
                'Content-Type': 'text/plain'
              }
            })
            .then((res) => {
              if (res.status === 201 || res.status === 200) {
                return axios(originalRequest);
              }
            });
        }
        break;
      default:
        return Promise.reject(error);
    }
  }
);

export default httpHandler;
