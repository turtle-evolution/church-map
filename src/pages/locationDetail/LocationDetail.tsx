import { useParams } from 'react-router-dom';

import {
  EuiPage,
  EuiPageContent,
  EuiEmptyPrompt,
  EuiPageBody,
  EuiLoadingSpinner
} from '@elastic/eui';

import { gql, useQuery } from '@apollo/client';

const LocationDetail = (): JSX.Element => {
  const { locationId } = useParams<{ locationId: string }>();

  const GET_CHURCH_DETAIL = gql`
    query GetChurchDetail($churchId: Int!) {
      churchs(where: { id: { _eq: $churchId } }) {
        name
        long_description
        address
      }
    }
  `;
  const { loading, error, data } = useQuery(GET_CHURCH_DETAIL, {
    variables: {
      churchId: locationId
    }
  });

  if (loading) {
    return (
      <div className="spinner-custom">
        <EuiLoadingSpinner size="l" />
      </div>
    );
  }
  if (error) {
    return <div>{error}</div>;
  }

  const dataDetail = data.churchs[0];
  return (
    <EuiPage paddingSize="none">
      <EuiPageBody>
        <EuiPageContent
          borderRadius="none"
          hasShadow={false}
          style={{ display: 'flex' }}
        >
          <EuiPageContent
            verticalPosition="center"
            horizontalPosition="center"
            paddingSize="none"
            color="subdued"
            hasShadow={false}
            data-testid="user-detail"
          >
            <EuiEmptyPrompt
              title={<span>{dataDetail.name}</span>}
              body={<>{dataDetail.long_description || dataDetail.address}</>}
            />
          </EuiPageContent>
        </EuiPageContent>
      </EuiPageBody>
    </EuiPage>
  );
};

export default LocationDetail;
