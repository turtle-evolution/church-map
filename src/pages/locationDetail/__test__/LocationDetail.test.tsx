import React from 'react';
import { render, screen } from '@testing-library/react';

import axios from 'axios';
import LocationDetail from '../LocationDetail';

jest.mock('mock/api');

jest.mock('react-router-dom', () => ({
  useParams: () => {
    return {
      userId: '1'
    };
  }
}));

jest.mock('axios');
const mockedAxios = axios as jest.Mocked<typeof axios>;

const fakeUser = {
  id: 1,
  name: 'Test User 1',
  address: {
    street: 'street'
  }
};

describe('LocationDetail', () => {
  test('should take a snapshot', () => {
    const { asFragment } = render(<LocationDetail />);

    expect(asFragment()).toMatchSnapshot();
  });

  test('should be render dataFake', async () => {
    mockedAxios.get.mockResolvedValue(fakeUser);
    render(<LocationDetail />);
    expect(screen.getByTestId('user-detail')).toContainHTML(
      '<div class="spinner-custom"><span class="euiLoadingSpinner euiLoadingSpinner--large" /></div>'
    );
  });
});
