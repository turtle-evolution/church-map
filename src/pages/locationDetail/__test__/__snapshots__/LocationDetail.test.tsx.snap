// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationDetail should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="euiPage euiPage--grow"
  >
    <div
      class="euiPageBody euiPageBody--borderRadiusNone"
    >
      <div
        class="euiPanel euiPanel--paddingLarge euiPanel--borderRadiusNone euiPanel--plain euiPanel--noShadow euiPageContent euiPageContent--borderRadiusNone"
        role="main"
        style="display: flex;"
      >
        <div
          class="euiPanel euiPanel--borderRadiusMedium euiPanel--subdued euiPanel--noShadow euiPanel--noBorder euiPageContent euiPageContent--verticalCenter euiPageContent--horizontalCenter"
          data-testid="user-detail"
          role="main"
        >
          <div
            class="spinner-custom"
          >
            <span
              class="euiLoadingSpinner euiLoadingSpinner--large"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
